service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -163,24 +170,16 @@ public final void processTenantMessage(final Message<JsonObject> tenantMsg) {
                 remove(tenantId, result -> reply(tenantMsg, result));
                 break;
             }
-        } catch (final IllegalArgumentException e) {
-            log.debug("invalid action in request message [{}]", e.getMessage());
-            reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
         } catch (final ClassCastException e) {
             log.debug("malformed request message [{}]", e.getMessage());
             reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
-        } catch (final NullPointerException e) {
-            log.debug("malformed request message [{}]", e.getMessage());
-            if (tenantMsg != null) {
-                reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
-            }
         }
     }
 
-    private void reply(final Message<JsonObject> request, final AsyncResult<TenantResult> result) {
-        if (result.succeeded()) {
-            if (result != null) {
-                reply(request, result.result());
+    private void reply(final Message<JsonObject> request, final AsyncResult<TenantResult> asyncActionResult) {
+        if (asyncActionResult.succeeded()) {
+            if (asyncActionResult != null) {
```

Comment:
this check doesn't make much sense, does it? I think you mean `asyncActionResult.result()` ...

Reply:
Of course, thanks for spotting.